src/error.c

```diff
@@ -579,19 +580,22 @@ Obj RequireArgumentEx(const char * funcname,
         strlcat(msgbuf, " ", sizeof(msgbuf));
     }
     strlcat(msgbuf, msg, sizeof(msgbuf));
-    if (IS_INTOBJ(op)) {
-        strlcat(msgbuf, " (not the integer %d)", sizeof(msgbuf));
-        arg1 = INT_INTOBJ(op);
-    }
-    else if (op == True)
-        strlcat(msgbuf, " (not the value 'true')", sizeof(msgbuf));
-    else if (op == False)
-        strlcat(msgbuf, " (not the value 'false')", sizeof(msgbuf));
-    else if (op == Fail)
-        strlcat(msgbuf, " (not the value 'fail')", sizeof(msgbuf));
-    else {
-        strlcat(msgbuf, " (not a %s)", sizeof(msgbuf));
-        arg1 = (Int)TNAM_OBJ(op);
+
+    if (showArgType) {
```

Comment:
With my `alternateArgType` from below, this could be replaced by the following (and no code would need to be reindented):
```C
    if (alternateArgType) {
        if (*alternateArgType) { // not the empty string?
            strlcat(msgbuf, " ", sizeof(msgbuf));
            strlcat(msgbuf, alternateArgType, sizeof(msgbuf));
        }
    } else if (IS_INTOBJ(op)) {
...
```

Reply:
I'll have another go. If I keep the full type (i.e. mutable list) , then also keeping the (not a...) becomes more reasonable too.